### ISSUE TITLE
Calendar ARIA

### ISSUE THREAD

**Issue by damyanpetev** [IgniteUI/igniteui-angular#6272]
## Is your feature request related to a problem? Please describe.
There's currently no ARIA attributes applied to the calendar (not in spec either). This makes screen readers announce partial information when navigating the calendar like "Clickable 27". 

## Describe the solution you'd like
The calendar should implement ARIA attributes per spec (see below) so each date can be announced when navigating with a screen reader.

## Describe alternatives you've considered
N/A

## Additional context
Per https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html the calendar should be using a grid pattern. Also some interesting abbreviation in the table headers.
Note:  This example might be outdated as it behaves a bit odd.
